src/private_cuda.h

```diff
@@ -128,4 +140,14 @@ struct _gpukernel {
 #endif
 };
 
+struct _gpucomm {
+  cuda_context* ctx;  // Start after the context
+// #ifdef WITH_CUDA_NCCL
+  ncclComm_t c;
+// #endif  // WITH_CUDA_NCCL
```

Comment:
I will search if the rest definition (gpucomm->ctx, where ctx is a `cuda_context`) is needed elsewhere and I will respond. Probably not, so you would be right. What about the `ncclResult_t` inside of `cuda_context`??